app/assets/javascripts/modules/accordion-with-descriptions.js

```diff
@@ -0,0 +1,160 @@
+(function (Modules) {
+  "use strict";
+
+  Modules.AccordionWithDescriptions = function() {
+
+    this.start = function($element) {
+
+      // Indicate that js has worked
+      $('.subsections').addClass('js-accordion-with-descriptions');
+
+       // Prevent FOUC, remove class hiding content
+      $('.subsections').removeClass('js-hidden');
+
+      // Insert the markup for the subsection-controls
+      $element.find('.subsection-controls')
+        .append( '<button class="js-open-close" aria-expanded="false">Open all</button>' );
+
+      // Get the open/close all button
+      var $openOrCloseAllButton = $element.find('.js-open-close');
+      var openOrCloseAllText = $openOrCloseAllButton.text();
+
+      // Wrap each title in a button, with aria controls matching the ID of the subsection
+      $element.find('.subsection-title').each(function(index) {
+        $(this).wrapInner( '<button class="subsection-button" aria-expanded="false" aria-controls="subsection_content_' + index +'"></a>' );
+      });
+
+      // Get all the sections
+      var totalSubsections = $element.find('.subsection-content').length;
+
+      // For each of the sections, create a string with all the subsection content IDs
+      var ariaControlsValue = "";
+
+      for (var i = 0; i < totalSubsections; i++) {
+        ariaControlsValue += "subsection_content_"+i+" ";
+      }
+
+      // Set the aria controls for the open/close all button value for all content items
+      var $openCloseButton = $element.find('.js-open-close');
+
+      $openCloseButton.attr('aria-controls', ariaControlsValue);
+
+      // Hide the content
+      $element.find('.subsection-content')
+        .addClass('js-hidden');
+
+      // Insert the subsection icon
+      $element.find('.subsection-header')
+        .append( '<span class="subsection-icon"></span>' );
+
+      // Add toggle functionality individual sections
+      $element.find('.subsection-header').on('click', function(e) {
+        toggleSection($(this).next());
+        toggleIcon($(this));
+        toggleState($(this).find('.subsection-button'));
+        setOpenCloseAllText();
+        return false;
+      });
+
+      $element.find('.subsection-button').on('click', function(e) {
+        toggleSection($(this).parent().parent().next());
+        toggleIcon($(this).parent().parent());
+        toggleState($(this));
+        setOpenCloseAllText();
+        return false;
+      });
```

Comment:
A wee one. Assigning `$(this)` to a variable and reusing will be a tiny bit more memory efficient, faster and more readable (for me at least). _shurg_